migrations/app/ddl_migrations/ddl_functions/20250513192427_fn_update_actual_progear_weight_totals.up.sql

```diff
@@ -0,0 +1,30 @@
+--B-22704 Michael Saki adding a function to update the progear and spousal progear actual weight totals in mto_shipments
```

Comment:
looks like you could simplify this by combining the subqueries into one select and also use a single update.

```suggestion
--B-22704 Michael Saki adding a function to update the progear and spousal progear actual weight totals in mto_shipments
CREATE OR REPLACE FUNCTION update_actual_progear_weight_totals(ppm UUID)
RETURNS void
LANGUAGE plpgsql AS $$
BEGIN
    UPDATE mto_shipments
    SET
        actual_pro_gear_weight = NULLIF(t.self_weight, 0),
        actual_spouse_pro_gear_weight = NULLIF(t.spouse_weight, 0)
    FROM (
        SELECT
            SUM(CASE WHEN pw.belongs_to_self = TRUE THEN pw.weight::int ELSE 0 END) AS self_weight,
            SUM(CASE WHEN pw.belongs_to_self = FALSE OR pw.belongs_to_self IS NULL THEN pw.weight::int ELSE 0 END) AS spouse_weight
        FROM progear_weight_tickets pw
        WHERE pw.ppm_shipment_id = ppm AND pw.deleted_at IS NULL
    ) t
    WHERE id = (SELECT shipment_id FROM ppm_shipments WHERE id = ppm);
END;
$$;
```

Reply:
Good catch! You chopped off my name though haha! 🤣

Reply:
5 hours later....🤣 -> [bb03806](https://github.com/transcom/mymove/pull/15442/commits/bb03806abdfe01d9092ac0645cc05c0231967dbd)